spec/controllers/dogs_controller_spec.rb

```diff
@@ -0,0 +1,116 @@
+require 'spec_helper'
+
+describe DogsController, type: :controller do
+
+  shared_examples 'an OK response with 1 dog' do
+    it 'responds successfully' do
+      expect(response).to be_success
+    end
+
+    it 'contains 1 dog' do
+      last_json = JSON.parse(response.body)
+      expect(last_json.length).to eq(1)
+    end
+
+    it 'contains found dog' do
+      expect([found_dog].to_json).to be_json_eql(response.body)
+    end
+  end
+
+  describe 'GET #index' do
+    context 'user is logged in' do
+      let!(:found_dog) { create(:dog, tracking_id: 1) }
+      let!(:other_dog) { create(:dog, tracking_id: 100) }
+      let(:params) { {} }
+
+      subject(:get_index) { get :index, params.merge(format: :json), {mgr_view: true} }
+
+      before do
+        allow(controller).to receive(:signed_in?).and_return(true)
+      end
+
+      context 'search by tracking id' do
+        let(:params) { { search: 1 } }
+
+        before do
+          get_index
+        end
+
+        it_behaves_like 'an OK response with 1 dog'
+      end
+
+      context 'search by name' do
+        let!(:found_dog) { create(:dog, name: 'oscar') }
+        let(:params) { { search: 'oscar' } }
+
+        before do
+          get_index
+        end
```

Comment:
since this is only one small short statement you could roll this block up into one inline proc. General pattern I've seen is - more than one line? do/end. One line statement, `{}`s are fine. Small coding style nitpick.